### ISSUE TITLE
multi_file_day testing

### ISSUE THREAD

**Issue by rstoneback** [pysat/pysat#266]
**Describe the bug**
There are no unit tests that engage the multi_file_day option.

**To Reproduce**

**Expected behavior**

Tests for this option to ensure it functions correctly now and into the future.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 N/A

**Additional context**
N/A

**Comment by jklenzing** [comment]
Should this be included as part of the test_instrument file?  I think that some of this is tested as part of the end-to-end tests of TIMED/SABER in the test_instruments.

**Comment by rstoneback** [comment]
Yes, I think we should have a suite of tests for multi-day using one of the test instruments.

I think we need another test class, with multi-day on, that goes through all of the normal tests we run on the Instrument object. Similar to xarray testing.

**Comment by jklenzing** [comment]
Linking to #512.

**Comment by aburrell** [comment]
I believe this has been completed.  @jklenzing @rstoneback do you concur?

**Comment by jklenzing** [comment]
I think the intent of this was to simulate stitching together multiple files for a day now that other instruments exist separately.  However, looking through coveralls, all the relevant code seems well covered.  Not sure if we need to add this.  @rstoneback?

**Comment by rstoneback** [comment]
> I think the intent of this was to simulate stitching together multiple files for a day now that other instruments exist separately. However, looking through coveralls, all the relevant code seems well covered. Not sure if we need to add this. @rstoneback?

I am behind on the latest state. @jklenzing you are correct the intent was to have a simulated instrument that would return data across day boundaries. We do have testing that engages the multi_file_day option and the code runs, however, it is not actually dealing with data across a day. The multi_file_day feature has been confirmed in practice but, last I checked, there was some potential for a latent bug.

**Comment by rstoneback** [comment]
See https://github.com/pysat/pysatNASA/issues/87 for a real-word encounter with a bug for xarray data formats.

**Comment by jklenzing** [comment]
Noting to myself that when I tried to use padding for the xarray instruments there, similar problems occurred.

**Comment by rstoneback** [comment]
> Noting to myself that when I tried to use padding for the xarray instruments there, similar problems occurred.

bummer

**Comment by jklenzing** [comment]
Potential solution: test instrument with a download routine that creates multiple files per day.  Make sure that boundaries do not match daybreak.

**Comment by jklenzing** [comment]
This is currently tested as part of the test_instrument_padding file, eg `TestMultiFileRightDataPaddingBasics`.  Is there something else that needs to be captured?

**Comment by jklenzing** [comment]
- [ ] add option for fake files to be generated over different time intervals
- [ ] add key to switch between multi_file_day modes
- [ ] identify problems and re-evaluate
